cmake/kokkos_tribits.cmake

```
@@ -231,16 +231,26 @@ endmacro()
 function(KOKKOS_SET_LIBRARY_PROPERTIES LIBRARY_NAME)
   cmake_parse_arguments(PARSE "PLAIN_STYLE" "" "" ${ARGN})
 
+  message(STATUS "Checking if linker can consume the Kokkos linker flags")
   if((NOT KOKKOS_ENABLE_COMPILE_AS_CMAKE_LANGUAGE) AND (${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.18"))
     #I can use link options
     #check for CXX linkage using the simple 3.18 way
+
+    #exclude case of compiler_launcher forwarding to nvcc_wrapper as the used CXX compiler is shadowed in this case (compiler_launcher changes the compiler).
+    #The CXX compiler CMake will invoke for the check is not able to consume the cuda flags.
+    if(NOT (KOKKOS_ENABLE_CUDA) OR ("${CMAKE_CXX_COMPILER}" MATCHES "nvcc_wrapper"))
```

We need to check the CMAKE_CXX_COMPILER vs `nvcc_wrapper` to not run this check with the compiler launcher (which would not be called by check_compiler_flags)

---

We normally check for `nvcc_wrapper` via
```suggestion
    if(NOT (KOKKOS_ENABLE_CUDA) OR (KOKKOS_CXX_COMPILER_ID STREQUAL NVIDIA))
```

---

as indicated above, this does not work as `check_compiler_flags` will call `CMAKE_CXX_COMPILER` and not the launcher

---

added in https://github.com/kokkos/kokkos/pull/7891/commits/58964409756c975e5f9de59ed63ebeb1a01accf6